src/wp-admin/includes/template.php

```diff
@@ -970,14 +970,22 @@ function parent_dropdown( $default_page = 0, $parent_page = 0, $level = 0, $post
  * Prints out option HTML elements for role selectors.
  *
  * @since 2.1.0
+ * @since 6.5.0 Added $roles_to_exclude parameter.
  *
- * @param string $selected Slug for the role that should be already selected.
+ * @param string $selected         Slug for the role that should be already selected.
+ * @param array  $roles_to_exclude Array of roles to exclude from the dropdown.
  */
-function wp_dropdown_roles( $selected = '' ) {
+function wp_dropdown_roles( $selected = '', $roles_to_exclude = array() ) {
```

Comment:
An alternative could be to not pass a list of roles to _exclude_, but the list of roles to _include_, defaulting to the editable roles.

Something like this:

```php
function wp_dropdown_roles( $selected = '', $editable_roles = null ) {
	if ( null === $editable_roles ) {
		$editable_roles = array_reverse( get_editable_roles() );
	}

	// ...
```

Then, the exclusion logic could be moved to `options-general.php`.

Curious to hear what others think about this.

Reply:
Cool idea, I changed the pr to use roles_to_include

Reply:
The way you implemented it is not really what I meant... Let me add some suggestions.

Reply:
Just added some more suggestions, hope the intention is clear.

Again, curious to hear what other contributors think